qemu/tests/block_copy.py

```diff
@@ -428,3 +428,44 @@ def verify_md5(self, file_name):
         if status != 0:
             raise error.TestFail("File %s changed, md5sum check output: %s" %
                                  (file_name, output))
+
+    def mount_disks(self):
+        """
+        Mount added disks inside guest.
+        """
+        os_type = self.params["os_type"]
+        if os_type == "linux":
+            session = self.get_session()
+            disks = utils_misc.list_linux_guest_disks(session)
+            mount_cmd = self.params["mount_cmd"]
+            for disk in disks:
+                if not re.search("da", disk):
+                    did = disk.split('/')[2]
+                    session.cmd_output_safe(mount_cmd % (did, did))
+
+    def format_disks(self):
+        """
+        Format all guest non-root disks, add file_names each file for each disk.
+        """
+        session = self.get_session()
+        os_type = self.params["os_type"]
+        file_names = self.params["file_names"]
+        if os_type == "linux":
+            all_disks_did = utils_misc.get_all_disks_did(session)
+            disks = utils_misc.list_linux_guest_disks(session)
+            for disk in disks:
+                if not re.search("da", disk):
+                    did = disk.split('/')[2]
+                    utils_misc.format_linux_disk(session, did, all_disks_did)
+                    # format_linux_disk() will mount disk automatically to /mnt/${did}
+                    self.params["file_names"] = "%s %s%s/%s" % (
+                        file_names, "/mnt/", did, "block_job_testfile")
+        elif os_type == "windows":
+            disk_num = len(self.params["images"].split())
+            for did in [1, disk_num - 1]:
```

Comment:
I am not sure if cases may need more than 3 images, but you will still got only 2 dids if there would be more than 3 images.
`for did in xrange(1, disk_num -1)` might be better. Or you may want to get did directly for the list(`self.params["images"].split()`).